twitter-borrowbot/services.py

```diff
@@ -96,32 +121,23 @@ def find_available_work(cls, book):
             if matches and matches["response"]["docs"]:
                 books = matches["response"]["docs"]
                 return next(book for book in books if book.get("openlibrary_work"))
-        except Exception:
-            print("Error fetching IA work")
-        return {}
+        except Exception as e:
+            raise FindAvailableWorkError(book=book, error=e)
 
 
 class Logger:
 
-    DELIMITER = "(:)>>--++--++--++--++--++--<<(:)"
-
-    @classmethod
-    def __init__(cls, tweet_filename, error_filename):
-        cls.tweet_filename = tweet_filename
-        cls.error_filename = error_filename
-
-    @classmethod
-    def log_tweet(cls, message):
-        f = open(cls.tweet_filename, "a")
+    @staticmethod
```

Comment:
Chances are, you don't want to be rolling your own Logger.

See: https://docs.python.org/3/howto/logging.html

Also, when opening a file, a lot can go wrong while a file is open and there are many ways you can be left in a messy state.

Python has "context managers" (see https://www.geeksforgeeks.org/context-manager-in-python) which define safer ways to perform tasks which require entering / leaving a context (such as entering a mode of opening a file).

The code pattern looks like this:

```
with open(filename, "a") as f:
    f.write(...)
 ```
 
As soon as an error is encountered or the `with` block is exited, the file will be cleaned up, because the file object defines an `__exit__` method (which is how the `with` function works) that defines how it should gracefully close / clean-up.

So:
1) Consider using python's built in `logger` module and for examples, see the logger setup for `openlibrary-client`: https://github.com/internetarchive/openlibrary-client/blob/master/olclient/openlibrary.py#L20
2) If you do have to manage your own file writing in python, you will almost never call f.close() because you'll be using a `with` and scoped context management